src/minisphere/event_loop.c

```diff
@@ -71,6 +126,65 @@ events_run_main_loop(void)
 	}
 }
 
+void
+events_tick(int api_version, bool clear_screen, int framerate)
+{
+	size_t              bytes_read;
+	struct socket_read* sr;
+
+	int i, len;
+
+	sphere_heartbeat(true, api_version);
+
+	if (!screen_skipping_frame(g_screen)) {
+		if (!dispatch_run(JOB_ON_RENDER))
+			return;
+	}
+	screen_flip(g_screen, framerate, clear_screen);
+	if (api_version >= 2)
+		image_set_scissor(screen_backbuffer(g_screen), screen_bounds(g_screen));
+
+	if (!dispatch_run(JOB_ON_UPDATE))
+		return;
+
+	// handle asynchronous socket reads
+	for (i = 0, len = vector_len(s_socket_reads); i < len; ++i) {
+		sr = vector_get(s_socket_reads, i);
+		bytes_read = socket_read(sr->socket, sr->ptr, sr->bytes_left);
+		sr->bytes_left -= bytes_read;
+		sr->ptr += bytes_read;
+		if (sr->bytes_left > 0 && !socket_connected(sr->socket)) {
+			jsal_push_ref_weak(sr->rejector);
+			jsal_push_new_error(JS_ERROR, "Connection was lost before completion of read");
+			jsal_call(1);
+			jsal_pop(1);
+			free_socket_read(sr);
+			vector_remove(s_socket_reads, i--);
+		}
+		else if (sr->bytes_left == 0) {
+			jsal_push_ref_weak(sr->resolver);
```

Comment:
You do an unnecessary check here - bytes_left is an unsigned int - it can only be == 0 or > 0

Swap the branches around so you do the == 0 completion path first then the second path only needs the !socket_connected check

Reply:
That makes sense.  It’s probably better for readability to have the normal success path first and the more exceptional conditions lower in the listing, anyway.  Thanks for the tip.

Reply:
Alright, I made the suggested change.  Agreed it's both cleaner this way and avoids the unnecessary extra check.